src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+    """
+
+    config_class = MaskRCNNConfig
+    main_input_name = "pixel_values"
+    supports_gradient_checkpointing = True
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, (nn.Linear, nn.Conv2d, nn.ConvTranspose2d)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, MaskRCNNForObjectDetection):
+            module.gradient_checkpointing = value
+
+
+MASK_RCNN_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`MaskRCNNConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+MASK_RCNN_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See
+            [`AutoImageProcessor.__call__`] for details.
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    """
+    Mask R-CNN Model (consisting of a backbone, region-proposal network (RPN) and RoI head) for object detection and
+    instance segmentation tasks.
+    """,
+    MASK_RCNN_START_DOCSTRING,
+)
+class MaskRCNNForObjectDetection(MaskRCNNPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.config = config
+
+        self.backbone = AutoBackbone.from_config(config.backbone_config)
+        self.neck = MaskRCNNFPN(config, self.backbone.channels)
+        self.rpn_head = MaskRCNNRPN(config)
+        self.roi_head = MaskRCNNRoIHead(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def parse_losses(self, losses):
+        """Parse the raw outputs (losses) of the network.
```

Comment:
Perhaps we could call this something like `aggregate_loss`? It's not really parsing anything, rather iterating and summing over dictionary values